lib/pluginmanager/update.rb

```diff
@@ -99,9 +100,25 @@ def plugins_to_update(previous_gem_specs_map)
       # If the plugins isn't available in the gemspec or in 
       # the gemfile defined with a local path, we assume the plugins is not
       # installed.
-      not_installed = plugins_arg.select{|plugin| !previous_gem_specs_map.has_key?(plugin.downcase) && !gemfile.find(plugin) }
+      not_installed = plugins_arg.select { |plugin| !previous_gem_specs_map.has_key?(plugin.downcase) && !gemfile.find(plugin) }
+
+      # find only the not installed that doesn't correspond to an alias
+      not_installed_aliases = not_installed.select { |plugin| LogStash::PluginManager::ALIASES.has_key?(plugin)}
+      not_installed -= not_installed_aliases
+
       signal_error("Plugin #{not_installed.join(', ')} is not installed so it cannot be updated, aborting") unless not_installed.empty?
-      plugins_arg
+
+      # resolve aliases that doesn't correspond to a real gem
+      plugins_to_update = plugins_arg.map do |plugin|
+        if not_installed_aliases.include?(plugin)
+          resolved_plugin = LogStash::PluginManager::ALIASES[plugin]
+          puts "Remapping alias #{plugin} to #{resolved_plugin}"
+          resolved_plugin
+        else
+          plugin
+        end
+      end
+      plugins_to_update
```

Comment:
minor optimization: when a user tries to update both an alias and its concrete class, this will ensure the concrete class is only updated once.

```suggestion
      plugins_to_update.uniq
```